### PR TITLE
[`peft`] Fix DP issues

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR fixes issues related to DP with `peft`

Adds also instructions on how to properly run DP + `peft`

cc @lvwerra

### CODE REVIEW

**Comment by lvwerra** [comment]
Looks good as a temporary fix but we should really change the API a bit to make this easier. :)

**Comment by younesbelkada** [comment]
I see, what you are suggesting is to simplify the model loading API a bit right?

**Comment by lvwerra** [comment]
Exactly, otherwise our API becomes more and more dark magic :D 

I think for NPP, PEFT, Int8 it should all become:

```python
model = AutoModelForCausalLMWithValueHead.from_pretrained(ckpt, method_specific_kwargs)
```
Internally we can then check if the kwargs are consistent and work in that combination and also have useful defaults for some of the approaches.

**Comment by younesbelkada** [comment]
Final training run for gpt2-peft in DP: https://wandb.ai/distill-bloom/trl/runs/anb919vh?workspace=user-younesbelkada

**Comment by lvwerra** [comment]
Looking good :)

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -461,15 +461,21 @@ def step(
                 model_inputs["decoder_attention_mask"] = self.accelerator.pad_across_processes(
                     model_inputs["decoder_attention_mask"], dim=1, pad_index=0, pad_first=pad_first
                 )
+            else:
+                model_inputs['labels'] = self.accelerator.pad_across_processes(
```
Comment (younesbelkada) [inline trl/trainer/ppo_trainer.py:465]: `labels` were used for decoder based models @lvwerra and I mistakenly deleted then in the previous PR

File: docs/source/sentiment_tuning_peft.mdx [hunk 2]
```diff
@@ -36,6 +36,24 @@ accelerate config # will prompt you to define the training configuration
...
+
+You can scale up to as many GPUs as you want, as long as you are able to fit the training process in a single device. The only tweak you need to apply is to load the model as follows:
+```python
+from accelerate import Accelerator
+...
+
+current_device = Accelerator().process_index
+
+pretrained_model = AutoModelForCausalLM.from_pretrained(
+    config.model_name, load_in_8bit=True, device_map={"": current_device}
```
Comment (lewtun) [inline docs/source/sentiment_tuning_peft.mdx:49]: Using an empty key in `device_map` seems a bit like magic to me - could we have a one-liner to explain why?

Comment (younesbelkada) [reply docs/source/sentiment_tuning_peft.mdx:49]: Sure yes, I will add it

File: trl/trainer/ppo_trainer.py (hunk 1 above)
Comment (lvwerra) [inline trl/trainer/ppo_trainer.py:465]: We actually don't need labels, right? In `prepare_model_inputs` we pop them for encoder-decoder, I think we should do the same for encoders there.

**Review by lvwerra — APPROVED** [review]
